Framework/Algorithms/src/SortXAxis2.cpp

```diff
@@ -21,6 +21,19 @@ const std::string SortXAxis::summary() const {
          "ascending or descending fashion.";
 }
 
+/** Validate the algorithm's properties.
+ * @return A map of property names and their issues.
+ */
+std::map<std::string, std::string> SortXAxis::validateInputs(void) {
+  std::map<std::string, std::string> issues;
+  MatrixWorkspace_sptr ws = getProperty("InputWorkspace");
+  if (!ws)
+    issues["InputWorkspace"] = "Cannot retrieve workspace";
+  if (ws->isHistogramData())
```

Comment:
What is the reason for checking histogram data and thus refusing to handle histogram data. The older Python one didn't do this and we already handle histogram specific data on line 134.

Reply:
An initial concern was the question if the documentation describes the algoritm correctly. (I agree, changing the documentation is easier and may be better here). Now, I see it possible to reverse the bin edges of histograms (from ascending to descending or descending to ascending). This happend in the corresponding test. Is it possible to create histograms with unsorted bin edges (CreateWorkspace)? Is this physically sensible to do? Imagine three bins with boundaries 4, 1, 9, 2: the first bin contains 5 counts, the second an third 0. What are the counts for three bins of boundaries 1, 2, 4, 9? In terms of input validation, one more requirement can be a numeric axis?

Reply:
After some discussion this afternoon I think I understand your problem better. I'll add back in support for Histograms but I will make sure that the histograms are proper histograms before the algorithm can start. Thus allowing for both Histograms to be supported and avoiding this error you are speaking off. It took some higher level explaining for me to get the actual gist of the problem.

Reply:
Sounds very good. You may be able to improve the algorithm and make it reliable. Happy that you do not only translate the Python algorithm into C++ (which is much easier and probably done). Thanks a lot.